src/goto-instrument/contracts/assigns.cpp

```diff
@@ -145,11 +166,11 @@ exprt assigns_clauset::generate_containment_check(const exprt &lhs) const
 exprt assigns_clauset::generate_subset_check(
   const assigns_clauset &subassigns) const
```

Comment:
This function needs to be updated for our "two sets" scheme now. We are only checking that:
`subassigns.write_set ⊆ write_set`.

I think we should be checking that:
`(subassigns.freely_assignable_set = {}) ∧ (subassigns.write_set ⊆ write_set ∪ freely_assignable_set)`

Reply:
Implemented as suggested.